### PR TITLE
afscompress: new port

### PR DESCRIPTION
**Pull request by ylluminarious**
#### Description

This adds a port for @RJVB's [fork of `afsctool`.](https://github.com/RJVB/afsctool) There is already a port for the original program by @brkirch, but this is a newer fork which adds several improvements. [Some folks tried getting this fork into Homebrew](https://github.com/Homebrew/homebrew-core/pull/20898) about 2 years ago, but as usual the maintainers of Homebrew made things into an unnecessary drama and rejected it. Then they proceeded to deride people over complaining about how difficult they were making the submission process... but I digress 😅. I have given this port a different name, [as per @brkirch's request.](https://github.com/Homebrew/homebrew-core/pull/20898#issuecomment-380727547) Hopefully we can have an easier time here than the folks at Homebrew 😄 

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.3 10G8

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? **(not applicable)**
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`? **(not applicable)**
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by RJVB** [comment]
I think I'd prefer if you could come up with a different name, like afsctool2 (I might even bump the version for that).

**Comment by ylluminarious** [comment]
@RJVB Sounds good to me. I came up with the name based on [another comment from](https://github.com/Homebrew/homebrew-core/pull/20898#issuecomment-380737553) @brkirch [on that thread.](https://github.com/Homebrew/homebrew-core/pull/20898#issuecomment-380737553) Do you think he'll be alright with such a name, i.e., `afsctool2`? Also, please let me know if/when you bump the version number so that I can update accordingly.

**Comment by RJVB** [comment]
@brkirch mentioned he is fine with `afsctool` being part of the name, just not the full name; `afsctool2` seems to comply with that.

**Comment by ylluminarious** [comment]
@RJVB Alright, then it's okay with me. Like I said, please let me know when you bump it so I can update it.

**Comment by RJVB** [comment]
On Sunday July 28 2019 11:18:51 George Plymale II wrote:
>@RJVB Alright, then it's okay with me. Like I said, please let me know when you bump it so I can update it.

There are a few loose ends I have to check before I do that, I'll *try* to do so this week.

**Comment by cjones051073** [comment]
> @RJVB Sounds good to me. I came up with the name based on [another comment from](https://github.com/Homebrew/homebrew-core/pull/20898#issuecomment-380737553) @brkirch [on that thread.](https://github.com/Homebrew/homebrew-core/pull/20898#issuecomment-380737553) Do you think he'll be alright with such a name, i.e., `afsctool2`? Also, please let me know if/when you bump the version number so that I can update accordingly.

afsctool2 would  not be a good choice for a port name, as given how we name ports this would tend to imply ‘version 2.x of afsctool’. It would be better to come up with something completely different.

**Comment by RJVB** [comment]
> this would tend to imply ‘version 2.x of afsctool’.

And that would be neither wrong nor at odds with brkirch's request.

**Comment by cjones051073** [comment]
> this would tend to imply ‘version 2.x of afsctool’.
> And that would be neither wrong nor at odds with brkirch's request.

Apart from the fact the version of your fork is 1.x not 2.x ...

Also, it implies `afsctool2` is related to `afsctool` in that the former is the 'official' version 2.x of the former. This is also not the case.

In my opinion `afsctool2` is not an appropriate name here.

**Comment by RJVB** [comment]
>Apart from the fact the version of your fork is 1.x not 2.x ...

Did you miss the fact that I'd bump the version?

>Also, it implies `afsctool2` is related to `afsctool` in that the former is the 'official' version 2.x of the former. This is also not the case.

It's not at all uncommon that software gets a version in its name to distinguish it from a previous version (and allow concurrent installation). I doubt that it is uncommon that this coincides with a change in maintainers/developers (in fact, I'd not be surprised if it's more common in cases where new people take over "abandonware").

I think that we wouldn't be having this debate if I myself had decided to change the application name when I got wind of @brkirch's request to change the name for distribution in HB.

Maybe he could chime in here...

I'm ready to bump the version (2.0 would make sense but in the end it's just a number) and I'm not unwilling to do an official rename to afsctool2. Anything else you'd have to do in the Portfile as long as you remember that I don't want to see my name nor my "fourcc" in it (you could use the numerical value of it, though 8-) )

**Comment by cjones051073** [comment]
> Apart from the fact the version of your fork is 1.x not 2.x ...
> Did you miss the fact that I'd bump the version?

That still would not make it the 'official' 2.x version of `afsctool`.

> Also, it implies `afsctool2` is related to `afsctool` in that the former is the 'official' version 2.x of the former. This is also not the case.
> It's not at all uncommon that software gets a version in its name to distinguish it from a previous version (and allow concurrent installation). I doubt that it is uncommon that this coincides with a change in maintainers/developers (in fact, I'd not be surprised if it's more common in cases where new people take over "abandonware"). I think that we wouldn't be having this debate if I myself had decided to change the application name when I got wind of @brkirch's request to change the name for distribution in HB. Maybe he could chime in here... I'm ready to bump the version (2.0 would make sense but in the end it's just a number) and I'm not unwilling to do an official rename to afsctool2. Anything else you'd have to do in the Portfile as long as you remember that I don't want to see my name nor my "fourcc" in it (you could use the numerical value of it, though 8-) )

I do not particular care what you call your github fork of `afsctool`. what I am requesting is the MacPorts port name for it needs to be more different to `afsctool` than just `afsctool2`.

**Comment by RJVB** [comment]
>That still would not make it the 'official' 2.x version of `afsctool`.

And it wouldn't be, it'd be the official 2.x version of the successor to afsctool. And even if it were, what's the problem? *I* have been maintaining the software for quite a while now, and can say in all modesty that my version has become something of the de-facto new official version.

Frankly, you can use that same argument against the original proposal too, "afsctool_rjvb" would imply that it's the "official RJVB version of afsctool". Which also wouldn't be wrong, it is indeed officially my version of that utility.

Initially I couldn't care less how MacPorts is going to call this software (with the aforementioned restrictions). Now, with this kind of headbutting I'm beginning to be of a mind to make similar demands as @brkirch and insist on a name of my choosing which in that case might well be `afscfool`.

**Comment by cjones051073** [comment]
>Frankly, you can use that same argument against the original proposal too, "afsctool_rjvb" would imply that it's the "official RJVB version of afsctool". Which also wouldn't be wrong, it is indeed officially my version of that utility.

No, my objection would not apply to `afsctool_rjvb`. It *specifically* applies to a name matching the pattern `<name><number>` which tends to mean something specifically within macports, that does not apply here.

> Initially I couldn't care less how MacPorts is going to call this software (with the aforementioned restrictions). Now, with this kind of headbutting I'm beginning to be of a mind to make similar demands as @brkirch and insist on a name of my choosing which in that case might well be `afscfool`.

I also do not care, as long as you avoid a name pattern such as above, which tends to mean something specific in MacPorts, that IMHO does not apply here.

Also, note, we are not homebrew. We are completely free to decide what software we package under what port names. So, we could if we wanted decide to completely replace the current `afsctool` port with your fork, if as you claim your fork has become the defacto default now. That should be discussed with the current maintainer of `afsctool`, @raimue, who incidently should also comment on the port naming used here.

**Comment by RJVB** [comment]
>So, we could if we wanted decide to completely replace the current `afsctool` port with your fork, if as you claim your fork has become the defacto default now. 

I have no problem with that, it's what I do with my own port.

However I do remember MacPorts at least taking into serious consideration objections from upstream about ports (VLC, openzfsonosx, ...).

**Comment by raimue** [comment]
I would not provide this fork with the name "afsctool" as the original author @brkirch specificially asked not to use this name for any fork. I think it is not unreasonable to respect this wish and give the fork another name.

Choosing another name would be up to @RJVB as upstream maintainer of the project. Probably it would be an even better idea to rename the upstream project according to this rule set by the original author. That would also make it clear this is now separately maintained.

Regardless of the name of the port, I agree with @cjones051073 that "afsctool2" even as the upstream project name could be misleading to be an official successor. How about "afsctool-ng"? Or has the "-ng" suffix for forks gone out of fashion already? 😉

I would also have no problem to remove the existing `afsctool` port after this new one has been merged. Development on the afsctool upstream is dead for years, while the fork offers the same functionality and more. Also it is more likely that this fork gets maintenance, overall it is a full replacement.

**Comment by mojca** [comment]
I would pick some name than's reasonably close and it might even resemble the original, but cannot be confused. Just appending `2` is probably not sufficient. Even if the original name only remains in description, it will still be super easy to find it. Any other prefix or suffix should be fine.

Either we should make it clear that it is a fork, or one should get official blessing for upstream to become the official maintainer / developer of the software in the future. Then it could easily be called with an appendix.

**Comment by RJVB** [comment]
I've pushed v1.7.0 to github:RJVB/afsctool . This includes a build option to use "PC" names (afscompress and zfscompress).

**Comment by ylluminarious** [comment]
@RJVB Thanks for letting me know, I've updated the Portfile accordingly.

**Comment by RJVB** [comment]
Can you change the PR title so I don't get a jolt each time a notification lands in my mailbox? :-/

**Comment by ylluminarious** [comment]
@RJVB As far as I know, it's not possible to change the name of a PR on Github. Do you mean to change the name of the port instead? As per @mojca's last comment and the discussion above that, I'm not yet sure what to change the port name to.

**Comment by cjones051073** [comment]
> @RJVB As far as I know, it's not possible to change the name of a PR on Github. Do you mean to change the name of the port instead? As per @mojca's last comment and the discussion above that, I'm not yet sure what to change the port name to.

If by 'name' you mean the PR description then yes it can be changed. Just hit the 'edit' button to the right of it...

**Comment by ylluminarious** [comment]
@cjones051073 Thanks for the tip. I don't think that's what he meant, though.

**Comment by cjones051073** [comment]
> @cjones051073 Thanks for the tip. I don't think that's what he meant, though.

I disagree. @RJVB asked for the 'PR title' to be changed, which is exactly this I would say.

**Comment by RJVB** [comment]
As per earlier comments, I do not wish to see my name nor initials in the portname. I'm going to presume that this will be respected; my current request indeed concerns the PR title which also ends up in the email notifications (`Re: [macports/macports-ports] afsctool_rjvb: new port (#4901)`).

The optional "politically correct" alternative names that can be obtained through my latest change are `afscompress` (and `zfscompress`), for Apple File System Compress. That harks back to one of the earlier UNIX compression utilities, called `compress` and adds the `afs` bit to refer to the fact it applies to Apple file systems (HFS and APFS).

My suggestion would be to call the port `afscompress` too; I assume that the port would have been called that way if I had actually renamed my fork before it was ever considered for inclusion in MacPorts.

R.

**Comment by ylluminarious** [comment]
@RJVB I apologize for causing you to be uncomfortable with the name. I'll rename it to `afscompress` as you suggest. I am also sorry about the PR title which is in your email notifications, I do not think I can change this.

**Comment by ylluminarious** [comment]
@mojca Thanks a lot for fixing the PR name!

**Comment by RJVB** [comment]
And my thanks to both of you.

No harm, no foul, but apologies accepted :)

**Comment by ylluminarious** [comment]
@RJVB Glad that's settled now 😄! I appreciate your courteousness.

I've made a few more minor updates to the Portfile to make it work right with your new options, so hopefully we can now get this approved and we'll be good to go.

**Comment by RJVB** [comment]
Well, I'm going to leave that to the powers that be ;)

File: sysutils/afscompress/Portfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    compress the specified files in parallel.
+
+checksums            sha256  911ae33960560e36d854753cca1fcb83bb4674123b83f04b5642980670bc747d \
+                     rmd160  b2ec4858164723532bee7d3b00f796d42aefbaea \
+                     size    110026
+
+depends_lib          port:sparsehash \
+                     port:zlib
+
+depends_build        port:cmake \
```
Comment (mf2k) [inline sysutils/afscompress/Portfile]: The cmake portgroup should be used instead.

File: sysutils/afscompress/Portfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+build {
+    system -W ${worksrcpath}       "mkdir ./build"
+    if {[variant_isset original_name]} {
+        system -W ${worksrcpath}/build "cmake -DCMAKE_INSTALL_PREFIX=${destroot}/${prefix} -Wno-dev .."
+    } else {
+        system -W ${worksrcpath}/build "cmake -DNEW_DRIVER_NAMES=ON -DCMAKE_INSTALL_PREFIX=${destroot}/${prefix} -Wno-dev .."
+    }
+    system -W ${worksrcpath}/build "make"
+}
```
Comment (mf2k) [inline sysutils/afscompress/Portfile]: Use the cmake portgroup instead.

Comment (ylluminarious) [reply sysutils/afscompress/Portfile]: @mf2k How do I pass in the correct build options with the `cmake` group? Also, do I have to replace the `github` group with `cmake`? If so, then I would have to do a lot of rigmarole to get it fetching the tarball from Github properly. Sadly, there's no documentation on the `cmake` group [in the MacPorts guide.](https://guide.macports.org/chunked/reference.portgroup.html)

**Comment by RJVB** [comment]
> @mf2k How ...

You can edit my own Portfile for this: https://github.com/RJVB/macstrop/blob/master/sysutils/afsctool/Portfile

Use 

```
github.setup    RJVB afsctool 1.7.0
name            afscompress
```

or (try which one works!)

```
github.setup    RJVB afsctool 1.7.0 v
name            afscompress
```

and remove the `fetch.type` and `livecheck' lines.

You'll need to add a checksums section but you will told what to use. You can add CMake options in `configure.args`, as shown on line 41/42.

R

File: sysutils/afscompress/Portfile (hunk 2 above)
Comment (mf2k) [reply sysutils/afscompress/Portfile]: You can use both portgroups. Best to look at other existing ports for guidance. Look at `vigra` for instance.

**Comment by ylluminarious** [comment]
@mf2k Thanks for approving this! And thanks for the advice on how to get the `cmake` group to work. And thank you for pointing me to your Portfile @RJVB. Thanks to everyone else who also participated in this PR.

**Comment by mf2k** [comment]
It looks like @raimue was not going to comment, so I went ahead and committed this. Thank you for your patience!

**Comment by ylluminarious** [comment]
@mf2k Thank you!

**Comment by mojca** [comment]
Please fix the port. It failed to build on all builders due to at least the missing `pkg-config` dependency.

**Comment by cjones051073** [comment]
Done. Also added cxx11 PG for atomic support.

Note, just FYI, port claims to support 10.6+, but in fact fails to build on <10.8

https://build.macports.org/builders/ports-10.7_x86_64_legacy-builder/builds/109096/steps/install-port/logs/stdio

**Comment by ylluminarious** [comment]
@mojca Sorry about that.

@mf2k Thanks for fixing that problem!

@cjones051073 Should I change the claim for 10.6? I have noticed other ports, too, failing on <10.8 and I'm not really sure why. I don't have any <10.8 machines or VMs handy so I can't test.

**Comment by cjones051073** [comment]
> @mojca Sorry about that.
> 
> @mf2k Thanks for fixing that problem!
> 
> @cjones051073 Should I change the claim for 10.6? I have noticed other ports, too, failing on <10.8 and I'm not really sure why. I don't have any <10.8 machines or VMs handy so I can't test.

Lets first wait and see what @RJVB says about the build failures.

Its not really a requirement to always put in a OS check for older OS builds that fail. Its nice to do it when it is known a port can never build on some OS, but often its just a matter of fixing some build requirement, like adding a compiler blacklist etc. This is exactly what c++11 PG does in affect, and it fixed the builds on 10.8 and 10.9 by forcing the use of a more up to date clang compiler than the default system one.

**Comment by cjones051073** [comment]
btw, the new overview site is really nice for looking at stuff like this, e.g.

https://ports.macports.org/port/afscompress/summary

**Comment by ylluminarious** [comment]
@cjones051073 Ok, thanks for the info. You're right about that new site too, looks really sexy and modern!

**Comment by RJVB** [comment]
>Lets first wait and see what @RJVB says about the build failures.

And he'll have to see what he can say about them. It's not unlikely that I've lost sight a bit of my goal to keep backwards compatibility by introducing C++ modernisms that are known (by some, not me ATM) to fail on 10.7 and earlier. I was surprised when I saw the mention of  atomic operations but then realised I may indeed have introduced them.

**Comment by cjones051073** [comment]
The remaining build failures are nothing to do with using a newer c++ standard (and yes, it does need c++1), but instead something to do with

```
/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_sysutils_afscompress/afscompress/work/afsctool-1.7.0/src/afsctool.h:26:10: fatal error: 'MacTypes.h' file not found
#include <MacTypes.h>
```

**Comment by RJVB** [comment]
>```
>/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_sysutils_afscompress/afscompress/work/afsctool-1.7.0/src/afsctool.h:26:10: fatal error: 'MacTypes.h' file not found
>#include <MacTypes.h>
>```

This is a bit surprising, what do we know about the availability of that file?

**Comment by cjones051073** [comment]
yes that works on macOS10.7

**Comment by RJVB** [comment]
That'd be Mac OS X 10.7 ;)

Thanks, fix pushed (a716a6a4341fdd5116c70114449c5377bca0c8c5).

**Pull request merged**